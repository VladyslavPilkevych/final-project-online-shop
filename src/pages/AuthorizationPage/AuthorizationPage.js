/* eslint-disable max-len */
import React, { memo } from 'react';
import { Formik, Form } from 'formik';
import * as yup from 'yup';
import { useDispatch } from 'react-redux';
import { useNavigate, Link } from 'react-router-dom';
import axios from 'axios';
import CustomInput from '../../components/CustomInput/CustomInput';
import { logInCustomer } from '../../api/api';
import { setUserTokenAC, getUser } from '../../store/actionCreators/userAC';
import styles from './AuthorizationPage.module.scss';

function AuthorizationPage() {
  const navigate = useNavigate();
  const dispatch = useDispatch();

  const initialValues = {
    loginOrEmail: '',
    password: '',
  };

  const onSubmit = async (values, { resetForm }) => {
    const response = await logInCustomer(values)
      .then((user) => user)
      .catch((err) => console.log(err));
    if (response && response.status === 200) {
      resetForm();
      dispatch(setUserTokenAC(response.data.token));
      dispatch(getUser(response.data.token));
      // localStorage.setItem('user', JSON.stringify(response.data.user));
      localStorage.setItem('token', JSON.stringify(response.data.token));
      navigate({ pathname: '/' });
    }
    // console.log(response);
  };

  const validationSchema = yup.object().shape({
    // login: yup.string().required('Поле обязательно').matches(/[A-Za-z ]/gi, 'Только латинские'),
    // email: yup.string().email('Неверный email').required('Поле обязательно'),
    password: yup.string().required('Поле обязательно'),
  });

  return (
    <div className={styles.authorizationPageWrapper}>
      <h1 className={styles.mainTitle}>Customer login</h1>
      <div className={styles.authorizationPageBlocksPageWrapper}>
        <div className={styles.authorizationBlockWrapper}>
          <h2 className={styles.customersBlocksTitle}>Registered Customers</h2>
          <p className={styles.customersBlocksSubtitle}>If you have an account, sign in with your login or email address</p>
          <Formik initialValues={initialValues} onSubmit={onSubmit} validationSchema={validationSchema}>
            <Form>
              <CustomInput name="loginOrEmail" type="text" placeholder="Email" />
              <CustomInput name="password" type="password" placeholder="Password" />
              <div>
                <button type="submit" className={styles.signInButton}>
                  Sign In
                </button>
              </div>
            </Form>
          </Formik>
        </div>
        <div className={`${styles.authorizationBlockWrapper} ${styles.signUpBlockWrapper}`}>
          <h2 className={styles.customersBlocksTitle}>New Customer?</h2>
          <p className={styles.customersBlocksSubtitle}>Creating an account has many benefits.</p>
          <ul className={styles.signUpBlockList}>
            <li>Check out faster</li>
            <li>Keep more than one address</li>
            <li>Track orders and more</li>
          </ul>
          <Link to="/sign-up">
            <button type="button" className={styles.signInButton}>
              Create an account
            </button>
          </Link>
        </div>
      </div>
    </div>
  );
}

export default memo(AuthorizationPage);
