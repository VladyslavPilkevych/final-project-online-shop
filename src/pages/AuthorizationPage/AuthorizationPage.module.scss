@import '../../assets/styles/global.module.scss';

.authorizationPageWrapper {
    max-width: 1398px;
    margin: 0 auto;
}

.mainTitle {
    font-size: 32px;
    font-weight: bold;
    font-family: $fonts;
    margin-top: 58px;
}

.authorizationPageBlocksPageWrapper {
    display: flex;
    column-gap: 34px;
    justify-content: center;
    margin-top: 21px;
}

.authorizationBlockWrapper {
    background-color: $bg-white-color;
    width: 35%;
    padding: 37px 60px 25px 60px;
}

.customersBlocksTitle {
    font-size: 18px;
    font-weight: 600;
    line-height: 27px;
}

.customersBlocksSubtitle {
    font-size: 14px;
    font-family: $fonts;
    font-style: normal;
    font-weight: 300;
    line-height: 30px;
}

.signInButton {
    margin-top: 30px;
    background-color: #2e58ff;
    color: $white;
    border-radius: 50px;
    padding: 10px;
    position: relative;
	display: inline-block;
    transition: 0.5s;
    border: none;
    // border: 5px solid $bg-white-color;
}
.signInButton:hover {
	background-color: darken(#1d4aff, 5%);
	box-shadow: 2px 2px 15px 5px #7b6aff;
	color: $white;
	cursor: pointer;
}

.signUpBlockList {
    list-style-type: disc;
    display: block;
    margin-top: 30px;
}

.signUpBlockList li {
    font-size: 14px;
    font-family: $fonts;
    font-style: normal;
    font-weight: 300;
    line-height: 24px;
}