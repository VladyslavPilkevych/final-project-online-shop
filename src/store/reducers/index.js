import { combineReducers } from 'redux';
import preloaderReducer from './preloaderReducer';
import productsReducer from './productsReducer';
import menuReducer from './menuReducer';
import userReducer from './userReducer';
import searchReducer from './searchReducer';
import cartReducer from './cartReducer';
import miniMenuReducer from './miniMenuReducer';
import filtersCategoriesReducer from './filtersCategoriesReducer';

const reducer = combineReducers({
  loader: preloaderReducer,
  products: productsReducer,
  menu: menuReducer,
  miniMenu: miniMenuReducer,
  filtersCategories: filtersCategoriesReducer,
  user: userReducer,
  search: searchReducer,
  cart: cartReducer,
});

export default reducer;
