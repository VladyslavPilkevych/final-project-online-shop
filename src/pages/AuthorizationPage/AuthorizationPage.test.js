import { render } from '@testing-library/react';
import AuthorizationPage from './AuthorizationPage';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from '../../store';

function Component() {
  return (
    <BrowserRouter>
      <Provider store={store}>
        <AuthorizationPage />
      </Provider>
    </BrowserRouter>
  );
}

describe('Snapshot AuthorizationPage should render', () => {
  test('Snapshot AuthorizationPage is rendered', () => {
    const { asFragment } = render(<Component />);
    expect(asFragment()).toMatchSnapshot();
  });
});