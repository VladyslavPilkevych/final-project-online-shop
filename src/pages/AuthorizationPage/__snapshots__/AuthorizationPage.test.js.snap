// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot AuthorizationPage should render Snapshot AuthorizationPage is rendered 1`] = `
<DocumentFragment>
  <div
    class="authorizationPageWrapper"
  >
    <h1
      class="mainTitle"
    >
      Customer login
    </h1>
    <div
      class="authorizationPageBlocksPageWrapper"
    >
      <div
        class="authorizationBlockWrapper"
      >
        <h2
          class="customersBlocksTitle"
        >
          Registered Customers
        </h2>
        <p
          class="customersBlocksSubtitle"
        >
          If you have an account, sign in with your login or email address
        </p>
        <form
          action="#"
        >
          <p
            class="inputLabel"
          >
            Email*
          </p>
          <input
            class="customInput"
            name="loginOrEmail"
            placeholder="Email"
            type="text"
            value=""
          />
          <p
            class="inputLabel"
          >
            Password*
          </p>
          <input
            class="customInput"
            name="password"
            placeholder="Password"
            type="password"
            value=""
          />
          <div>
            <button
              class="signInButton"
              type="submit"
            >
              Sign In
            </button>
          </div>
        </form>
      </div>
      <div
        class="authorizationBlockWrapper signUpBlockWrapper"
      >
        <h2
          class="customersBlocksTitle"
        >
          New Customer?
        </h2>
        <p
          class="customersBlocksSubtitle"
        >
          Creating an account has many benefits.
        </p>
        <ul
          class="signUpBlockList"
        >
          <li>
            Check out faster
          </li>
          <li>
            Keep more than one address
          </li>
          <li>
            Track orders and more
          </li>
        </ul>
        <a
          href="/sign-up"
        >
          <button
            class="signInButton"
            type="button"
          >
            Create an account
          </button>
        </a>
      </div>
    </div>
    <div
      class="Toastify"
    />
  </div>
</DocumentFragment>
`;
